frontends/election-manager/src/components/print_all_ballots_button.tsx

```diff
@@ -0,0 +1,362 @@
+      addPrintedBallot({
+        ballotStyleId,
+        precinctId,
+        locales: defaultBallotLocales,
+        numCopies: ballotCopies,
+        printedAt: new Date().toISOString(),
+        type,
+      });
+    }
+  }, [
+    addPrintedBallot,
+    ballotCopies,
+    ballotIndex,
+    ballotMode,
+    ballotStyles,
+    isAbsentee,
+    logger,
+    userRole,
+  ]);
+
+  const advancePrinting = useCallback(async () => {
+    assert(typeof ballotIndex === 'number');
+    if (ballotIndex < ballotStyles.length - 1) {
+      await makeCancelable(
+        sleep(PRINTER_WARMUP_TIME + ballotCopies * TWO_SIDED_PRINT_TIME)
+      );
+      setBallotIndex(ballotIndex + 1);
+    } else {
+      // For the last print job, rather than waiting for all pages to finish
+      // printing, free up the UI from the print modal
+      await makeCancelable(sleep(PRINTER_COOLDOWN_TIME));
+      closeModal();
+    }
+  }, [
+    ballotCopies,
+    ballotIndex,
+    ballotStyles.length,
+    closeModal,
+    makeCancelable,
+  ]);
+
+  const onPrintError = useCallback(async () => {
+    setPrintFailed(true);
+    setModalState('no-printer');
+    setBallotIndex(undefined);
+    await logger.log(LogEventId.BallotPrinted, userRole, {
+      disposition: 'failure',
+      message:
+        'Failed to print ballots while printing all ballot styles because printer was disconnected.',
+      result: 'User directed to reconnect printer.',
+      error: 'Printer disconnected.',
+    });
+  }, [logger, userRole]);
+
+  const onRendered = useCallback(async () => {
+    assert(typeof ballotIndex === 'number');
+    if (printLock.lock()) {
+      if (needsPrinter) {
+        printLock.unlock();
+        return onPrintError();
+      }
+      await printer.print({
+        sides: 'two-sided-long-edge',
+        copies: ballotCopies,
+      });
+      await logBallotsPrinted();
+      await advancePrinting();
+      printLock.unlock();
+    }
+  }, [
+    ballotIndex,
+    printLock,
+    needsPrinter,
+    printer,
+    ballotCopies,
+    logBallotsPrinted,
+    advancePrinting,
+    onPrintError,
+  ]);
+
+  let mainContent = null;
+  let actions = null;
+  let onOverlayClick: VoidFunction | undefined = closeModal;
+
+  switch (modalState) {
+    case 'no-printer':
+      mainContent = (
+        <Prose>
+          <h1>
+            {printFailed ? 'Printer Disconnected' : 'No Printer Detected'}
+          </h1>
+          <p>
+            {printFailed
+              ? 'Printing stopped because the printer was disconnected. Reconnect the printer to try again.'
+              : 'Please connect the printer to print all ballot styles.'}
+          </p>
+        </Prose>
+      );
+      actions = <LinkButton onPress={closeModal}>Cancel</LinkButton>;
+      break;
+
+    case 'options':
+      mainContent = (
+        <Prose>
+          <h1>Print All Ballot Styles</h1>
+          <p>Select the ballot type and number of copies to print:</p>
+          <CenteredOptions>
```

Comment:
Instead of using custom styles, is there a prop higher up that we could rely on, e.g. the `Modal` `centerContent` prop or the `Prose` `textCenter` prop?

Reply:
It seemed weird to me to wrap a `Prose` in a `Prose`, but that would do the trick. Would you prefer that?

Reply:
Ah yes, I totally agree that a `Prose` in a `Prose` doesn't make sense - I was actually wondering if we could apply `textCenter` to the existing outer `Prose`

That said, I'm remembering now that the header and "Select the ballot type" prompt are currently left aligned. If we're okay with centering everything (I think that would look okay but understand if that's not what we want), then that should work! If we intentionally want those elements left aligned, then the current solution is fine!

Reply:
For no particularly good reason (other than perhaps being consistent with the other modals in the app) I prefer the left-align, so I will leave it as is.

Reply:
> other than perhaps being consistent with the other modals in the app

^ I think that's a great reason!